sources/Renderer/Direct3D12/Shader/D3D12Shader.cpp

```diff
@@ -232,6 +235,30 @@ void D3D12Shader::BuildStreamOutput(UINT numVertexAttribs, const VertexAttribute
     }
 }
 
+bool IsProfileDxcAppropriate(const char* target)
+{
+    size_t szLen = StrLength(target);
+
+    // LLGL allows for a blank string to be sent into the target of a ShaderDescriptor, but
+    // FXC nor DXC supports this behavior, so it doesn't matter if we send to FXC or DXC.
+    if (szLen == 0)
+        return false;
+
+    for_range(i, szLen)
```

Comment:
`StrLength` relies on the string being NUL-terminated, just like `strlen`, so this loop could combine the search for the end and the `_` character. Perhaps something like this including the explanation of what we are parsing:
```cpp
// Search for first occurrence of '_' to parse the FXC/DXC SM pattern '[lib|vs|ps|...]_D_D', e.g. 'vs_6_0'
for (const char* c = target; c != '\0'; ++c)
{
    if (*c == '_')
    {
        // Safe to use next character in NUL-terminated string, since it's either our major version number or '\0'
        char majorShaderVersion = c[1];
        if (majorShaderVersion >= '6')
            return true;
    }
}
```